source/FreeRTOS_TCP_State_Handling_IPV4.c

```diff
@@ -167,13 +174,8 @@ FreeRTOS_Socket_t * prvHandleListen_IPV4( FreeRTOS_Socket_t * pxSocket,
         const ProtocolHeaders_t * pxProtocolHeaders = ( ( const ProtocolHeaders_t * )
                                                         &( pxNetworkBuffer->pucEthernetBuffer[ ipSIZE_OF_ETH_HEADER + uxIPHeaderSizePacket( pxNetworkBuffer ) ] ) );
 
-        if( pxNetworkBuffer->pxEndPoint != NULL )
-        {
-            pxReturn->pxEndPoint = pxNetworkBuffer->pxEndPoint;
-        }
-
-        configASSERT( pxReturn->pxEndPoint != NULL );
-
+        /* The endpoint in network buffer must be valid in this condition. */
+        pxReturn->pxEndPoint = pxNetworkBuffer->pxEndPoint;
         pxReturn->bits.bIsIPv6 = pdFALSE_UNSIGNED;
```

Comment:
Can we keep the `configASSERT( pxReturn->pxEndPoint != NULL );` here. It looks like `pxNetworkBuffer->pxEndPoint;` can be NULL.

Reply:
No, it's impossible because pxNetworkBuffer->pxEndPoint is checked at the begining of function.
```
pxEndpoint = pxNetworkBuffer->pxEndPoint;

if( ( pxEndpoint != NULL ) && ( pxTCPPacket->xIPHeader.ulDestinationIPAddress == pxEndpoint->ipv4_settings.ulIPAddress ) )
```

Reply:
Sure, thanks